src/tests/public-holidays.service.mock.test.ts

```diff
@@ -0,0 +1,75 @@
+import axios from 'axios';
+import {
+  getListOfPublicHolidays,
+  checkIfTodayIsPublicHoliday,
+  getNextPublicHolidays
+} from '../services/public-holidays.service';
+import { holidaysList2023, nextPublicHolidays } from './mock';
+import { PublicHolidayStatus } from '../helpers';
+
+
+describe('testing services by mock', () => {
+  it('should return public holidays list (getListOfPublicHolidays)', async () => {
+    jest.spyOn(axios, 'get').mockImplementation(() => Promise.resolve({ data: holidaysList2023 }));
```

Comment:
FYI: for axios mocking you can use https://www.npmjs.com/package/axios-mock-adapter

Reply:
axios-mock-adapter applied once.